src/schrodinger/sketcher/dialog/paste_in_text_dialog.cpp

```diff
@@ -1,10 +1,88 @@
 #include "paste_in_text_dialog.h"
 
+#include <algorithm>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "schrodinger/rdkit_extensions/convert.h"
+#include "schrodinger/rdkit_extensions/file_format.h"
 #include "schrodinger/sketcher/model/sketcher_model.h"
 #include "schrodinger/sketcher/ui/ui_paste_in_text_dialog.h"
 
+using ::schrodinger::rdkit_extensions::Format;
+
 namespace schrodinger
 {
+
+namespace
+{
+
+typedef uint8_t FormatMoleculeType_t;
+namespace FormatMoleculeType
+{
+/**
+ * Whether a given format applies to atomistic models, monomeric models, or both
+ */
+enum : FormatMoleculeType_t {
+    ATOMISTIC = 1 << 0,
+    MONOMERIC = 1 << 1,
+    BOTH = ATOMISTIC | MONOMERIC,
+};
+} // namespace FormatMoleculeType
+
+/**
+ * All formats to load into the format combo box, and whether they apply to
+ * atomistic or monomeric models
+ */
+const std::vector<std::tuple<QString, Format, FormatMoleculeType_t>> FORMATS = {
+    {"Autodetect", Format::AUTO_DETECT, FormatMoleculeType::BOTH},
+    {"RDKit Base64", Format::RDMOL_BINARY_BASE64, FormatMoleculeType::BOTH},
+    {"SMILES", Format::SMILES, FormatMoleculeType::ATOMISTIC},
+    {"Extended SMILES", Format::EXTENDED_SMILES, FormatMoleculeType::ATOMISTIC},
+    {"SMARTS", Format::SMARTS, FormatMoleculeType::ATOMISTIC},
+    {"Extended SMARTS", Format::EXTENDED_SMARTS, FormatMoleculeType::ATOMISTIC},
+    {"MDL SD V3000", Format::MDL_MOLV3000, FormatMoleculeType::ATOMISTIC},
+    {"MDL SD V2000", Format::MDL_MOLV2000, FormatMoleculeType::ATOMISTIC},
+    {"Maestro", Format::MAESTRO, FormatMoleculeType::ATOMISTIC},
+    {"InChI", Format::INCHI, FormatMoleculeType::ATOMISTIC},
+    {"PDB", Format::PDB, FormatMoleculeType::ATOMISTIC},
+    {"MOL2", Format::MOL2, FormatMoleculeType::ATOMISTIC},
+    {"XYZ", Format::XYZ, FormatMoleculeType::ATOMISTIC},
+    {"Marvin Document", Format::MRV, FormatMoleculeType::ATOMISTIC},
+    {"ChemDraw XML", Format::CDXML, FormatMoleculeType::ATOMISTIC},
+    {"HELM", Format::HELM, FormatMoleculeType::MONOMERIC},
+    {"FASTA Peptide", Format::FASTA_PEPTIDE, FormatMoleculeType::MONOMERIC},
+    {"FASTA DNA", Format::FASTA_DNA, FormatMoleculeType::MONOMERIC},
+    {"FASTA RNA", Format::FASTA_RNA, FormatMoleculeType::MONOMERIC},
```

Comment:
Hm, could you instead update and use this function? https://github.com/schrodinger/sketcher/blob/68cbcd5ab7fc5cb4e3410e449b7e2fcfde7a1322/src/schrodinger/sketcher/dialog/file_import_export.cpp#L17
You'd have to add SMARTS, HELM, and the various FASTAs

(and update this call site to skip anything without an extension)
https://github.com/schrodinger/sketcher/blob/68cbcd5ab7fc5cb4e3410e449b7e2fcfde7a1322/src/schrodinger/sketcher/widget/sketcher_top_bar.cpp#L245